src/main/java/org/alfresco/util/ScriptUtil.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.util;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+
+import org.springframework.core.io.support.EncodedResource;
+
+public abstract class ScriptUtil
+{
+    private static final String DEFAULT_SCRIPT_COMMENT_PREFIX = "--";
+
+    /**
+     * Read a script from the provided EncodedResource and build a String containing
+     * the lines.
+     *
+     * @param resource
+     *            the resource (potentially associated with a specific encoding) to
+     *            load the SQL script from
+     * @return a String containing the script lines
+     */
+    public static String readScript(EncodedResource resource) throws IOException
+    {
+        return readScript(resource, DEFAULT_SCRIPT_COMMENT_PREFIX);
+    }
+
+    /**
+     * Read a script from the provided EncodedResource, using the supplied line
+     * comment prefix, and build a String containing the lines.
+     *
+     * @param resource
+     *            the resource (potentially associated with a specific encoding) to
+     *            load the SQL script from
+     * @param lineCommentPrefix
+     *            the prefix that identifies comments in the SQL script (typically
+     *            "--")
+     * @return a String containing the script lines
+     */
+    private static String readScript(EncodedResource resource, String lineCommentPrefix) throws IOException
+    {
+        LineNumberReader lineNumberReader = new LineNumberReader(resource.getReader());
+        try
+        {
+            return readScript(lineNumberReader, lineCommentPrefix);
+        }
+        finally
+        {
+            lineNumberReader.close();
+        }
+    }
+
+    /**
+     * Read a script from the provided LineNumberReader, using the supplied line
+     * comment prefix, and build a String containing the lines.
+     *
+     * @param lineNumberReader
+     *            the LineNumberReader containing the script to be processed
+     * @param lineCommentPrefix
+     *            the prefix that identifies comments in the SQL script (typically
+     *            "--")
+     * @return a String containing the script lines
+     */
+    private static String readScript(LineNumberReader lineNumberReader, String lineCommentPrefix) throws IOException
+    {
+        String statement = lineNumberReader.readLine();
+        StringBuilder scriptBuilder = new StringBuilder();
+        while (statement != null)
+        {
+            if (lineCommentPrefix != null && !statement.startsWith(lineCommentPrefix))
```

Comment:
It might be better to either extract and reuse the code from SchemaBootstrap#executeScriptFile (it already handles comments, variables, loops and etc) or just remove the comment handling from here at all to avoid duplication.

Reply:
The SchemaBootstrap#executeScriptFile provides extensive functionality and is tightly coupled. I wouldn't add that part because this would allow users to do a lot more that they should with this file.

Reply:
ok, fair enough.